src/sonic-config-engine/minigraph.py

```diff
@@ -103,7 +104,83 @@ def parse_device(device):
             deployment_id = node.text
     return (lo_prefix, lo_prefix_v6, mgmt_prefix, name, hwsku, d_type, deployment_id)
 
-def parse_png(png, hname):
+def calculate_lcm_for_ecmp (nhdevices_bank_map, nhip_bank_map):
+    banks_enumerated = {}
+    lcm_array = []
+    for value in nhdevices_bank_map.values():
+        for key in nhip_bank_map.keys():
+            if nhip_bank_map[key] == value:
+                if value not in banks_enumerated:
+                    banks_enumerated[value] = 1
+                else:
+                    banks_enumerated[value] = banks_enumerated[value] + 1
+    for bank_enumeration in banks_enumerated.values():
+        lcm_list = range(1, bank_enumeration+1)
+        lcm_comp = lcm_list[0]
+        for i in lcm_list[1:]:
+            lcm_comp = lcm_comp * i / gcd(lcm_comp, i)
+        lcm_array.append(lcm_comp)
+
+    LCM = sum(lcm_array)
+    return LCM
+
+def formulate_ecmp_entry(dpg_ecmp_content, port_device_map):
+    FG_NHG_MEMBER = [{}, {}]
+    FG_NHG_PREFIX = [{}, {}]
+    FG_NHG = [{}, {}]
+    FG_NEIGH = [{}, {}]
+    COMP_FG_NHG_MEMBER = {}
+    COMP_FG_NHG_PREFIX = {}
+    COMP_FG_NHG = {}
+    COMP_FG_NEIGH = {}
+    neigh_key_ipv4 = []
+    neigh_key_ipv6 = []
+    ipv4_tag = "fgnhg_v4"
+    ipv6_tag = "fgnhg_v6"
+    port_nhipv4_map = dpg_ecmp_content[0]
+    port_nhipv6_map = dpg_ecmp_content[1]
+    nhgaddr = dpg_ecmp_content[2]
+    nhg_int = dpg_ecmp_content[3]
```

Comment:
nit: readability consideration, I would recommend making dpg_ecmp_content a dictionary instead of an array.